### ISSUE TITLE
Problem with --num-duplex-prep-workers option in duplex_from_pod5_and_bam 

### ISSUE THREAD

**Issue by santanaw** [nanoporetech/remora#119]
Hi @marcus1487 . I have been trying to run `remora infer duplex_from_pod5_and_bam`  using the GPU as shown below.
```
remora infer duplex_from_pod5_and_bam \
my.pod5 \
my.movetab.bam \
my.dx.bam \
pair_ids_filtered.txt \
--model ${REMORA2_DIR}/dna_r10.4.1_e8.2_4khz_400bps_sup_v4.1.0_5hmc_5mc_CG_v2.pt \
--out-bam my.dx.mod.bam \
--log-filename my.dx.mod.log \
--device 0 \
--num-extract-alignment-workers 2 \
--num-infer-workers 2 \
--num-duplex-prep-workers 2 
```
However each time I set the option `--num-duplex-prep-workers` to something different than 1, I receive the next type of errors in the log:
```
DEBUG [21:09:31:MakeDuplexReads_1:MainThread:util.py:417] UNEXPECTED_ERROR in MakeDuplexReads worker: 'truncated file'.
Full traceback: Traceback (most recent call last):
  File "/anaconda3/envs/remora_v2.1.2/lib/python3.11/site-packages/remora/util.py", line 414, in _mt_func
    out_q.put(func(val, *args, **kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/anaconda3/envs/remora_v2.1.2/lib/python3.11/site-packages/remora/inference.py", line 520, in make_duplex_reads
    bam_record = next(bam_file_handle)
                 ^^^^^^^^^^^^^^^^^^^^^
  File "pysam/libcalignmentfile.pyx", line 1875, in pysam.libcalignmentfile.AlignmentFile.__next__
OSError: truncated file
```
and in the stdout:
```
60it [00:21, 4.44it/s][E::sam_parse1] SEQ and QUAL are of different length
[W::sam_read1_sam] Parse error at line 25
```
Is there any way to make this work? So far, changing the `--num-duplex-prep-workers`  to more than 1 is the only way I could see a better runtime performance on my calls. Otherwise I get something around 1it/s. Or is there an alternative to improve performance?

Thank you for any information

**Comment by marcus1487** [comment]
We are looking into this issue internally. It seems that the pysam file handles do not play well with multithreading. We will try to get a fix for this into the next release, but keep this thread updated either way.

**Comment by santanaw** [comment]
Many thanks for looking into this.

**Comment by marcus1487** [comment]
This issue should be resolved in Remora 3.0. Please re-open this issue if you continue to experience issues.
